lib/Sema/CSDiag.cpp

```diff
@@ -5241,14 +5243,31 @@ bool FailureDiagnosis::visitApplyExpr(ApplyExpr *callExpr) {
       (isUnresolvedOrTypeVarType(fnExpr->getType()) ||
        (fnExpr->getType()->is<AnyFunctionType>() &&
         fnExpr->getType()->hasUnresolvedType()))) {
+    // FIXME: Prevent typeCheckChildIndependently from transforming expressions,
```

Comment:
Yes, that's exactly right, the same situation as with InOutExpr from https://github.com/apple/swift/pull/5105 , once typeCheckChildIndependency is improved, we can remove this workaround.